src/dvclive/live.py

```diff
@@ -41,95 +41,135 @@ def __init__(
+        os.makedirs(self.dir, exist_ok=True)
 
-        if self.report_mode in ("html", "md"):
-            if not self.report_file:
-                self.report_file = os.path.join(self.dir, f"report.{report}")
-            out = Path(self.report_file).resolve()
-            logger.info(f"Report file (if generated): {out}")
+        self._report_mode: Optional[str] = report
+        self._init_report()
 
         if self._resume:
-            self._read_params()
-            self._step = self.read_step()
-            if self._step != 0:
-                self._step += 1
-            logger.info(f"Resumed from step {self._step}")
+            self._init_resume()
         else:
-            self._cleanup()
+            self._init_cleanup()
+
+        self._baseline_rev: Optional[str] = None
+        self._exp_name: Optional[str] = None
+        self._inside_dvc_exp: bool = False
+        self._dvc_repo = None
+        self._init_dvc()
 
+        self._studio_url: Optional[str] = None
+        self._studio_token: Optional[str] = None
         self._latest_studio_step = self.step if resume else -1
-        if self.report_mode == "studio":
-            from scmrepo.git import Git
+        self._studio_events_to_skip: Set[str] = set()
+        self._init_studio()
 
-            self.rev = Git().get_rev()
+    def _init_resume(self):
+        self._read_params()
+        self._step = self.read_step()
+        if self._step != 0:
+            self._step += 1
+        logger.debug(f"{self._step=}")
 
-            if not post_to_studio(self, "start", logger):
-                logger.warning(
-                    "`post_to_studio` `start` event failed. "
-                    "`studio` report cancelled."
-                )
-                self.report_mode = None
-
-    def _cleanup(self):
+    def _init_cleanup(self):
         for plot_type in PLOT_TYPES:
             shutil.rmtree(
                 Path(self.plots_dir) / plot_type.subfolder, ignore_errors=True
             )
 
         for f in (self.metrics_file, self.report_file, self.params_file):
-            if os.path.exists(f):
+            if f and os.path.exists(f):
                 os.remove(f)
 
-    def _init_paths(self):
-        os.makedirs(self.dir, exist_ok=True)
+    def _init_dvc(self):
+        self._dvc_repo = get_dvc_repo()
+        if os.getenv(env.DVC_EXP_BASELINE_REV, None):
+            # `dvc exp` execution
+            self._baseline_rev = os.getenv(env.DVC_EXP_BASELINE_REV, "")
+            self._exp_name = os.getenv(env.DVC_EXP_NAME, "")
+            self._inside_dvc_exp = True
+        else:
+            # `Python Only` execution
+            # TODO: How to handle `dvc repro` execution?
+            if self._dvc_repo is not None:
+                self._baseline_rev = self._dvc_repo.scm.get_rev()
+                self._exp_name = random_exp_name(
+                    self._dvc_repo, self._baseline_rev
+                )
+                make_dvcyaml(self)
```

Comment:
I'm not as sure that we should skip `make_dvcyaml`, but we can follow up on this separate from this PR.

Reply:
Another follow up: pass an experiment name to use.

Reply:
And another: consider whether to include the dvclive user script/notebook in the tracked files.

Reply:
> And another: consider whether to include the dvclive user script/notebook in the tracked files.

Do you mean to include it in the `include_tracked` list passed to experiments save or other thing?

Reply:
Yes, whether to include it in `include_tracked`. I'm not sure it's needed, so it was more of a discussion point than a request. There's something clean about the default being to only save the live dir.